wordpress-mobile/WordPress-iOS
# [Gutenberg] - PR 02 - Editor default and migration

Third step to implement https://github.com/wordpress-mobile/gutenberg-mobile/issues/1233
Based on https://github.com/wordpress-mobile/WordPress-iOS/pull/12166

This PR implements the editor default logic and the migration of the local setting to remote, when the remote setting is empty.

To test:
- Open the test user's report card
  - Check that nothing is set on `SELECTED EDITORS` for mobile platform.
  - You can press `reset` to remove the setting if there's any.

**Sync and Migration from local to remote:**
- Login with a test user.
  - Check that the locally defined default editor was set on the report card.
- Toggle the switch 
  - Check that the new value was set on the report card.

**Sync from remote to local**
- Set a blog on Aztec and a blog on Gutenberg
- Uninstall the app, install and login again
- Check that the editor switch correspond to the editor set on the report card for both blogs

**Auto-enabled dialog**
- Open a Gutenberg post on a blog originally set to use Aztec (from a clean install)
  - Check that the dialog appears and both the switch and the report card is set to gutenberg for that blog
- Open the other blog set to gutenberg.
- Switch it back to Aztec.
- Open a blog with gutenberg blocks.
  - Check that it opens with Gutenberg, but the switch and the report card remain as Aztec, and no dialog appears.

**Migration from global setting to per-site setting**
- On 12.9 set to Gutenberg:
  - Updating to 13.0, all blogs should default to gutenberg
  - No autoenable-dialog should appear for any blog.

- On 12.9 set to Aztec:
  - Updating to 13.0 should: 
    - Keep Aztec on older users (most users)
    - Auto-enable gutenberg for newer users

**New installs from 13.0 and beyond**
- The default editor should be Gutenberg for all newer users
- The default editor should be Aztec for all older users

**Note:** A newer user is defined as having a userID > 149287441

cc @daniloercoli
